awx/ui_next/src/App.jsx

```diff
@@ -29,53 +30,51 @@ const ProtectedRoute = ({ children, ...rest }) =>
   );
 
 function App() {
-  const catalogs = { en, ja };
   let language = getLanguageWithoutRegionCode(navigator);
-  if (!Object.keys(catalogs).includes(language)) {
+  if (!Object.keys(locales).includes(language)) {
     // If there isn't a string catalog available for the browser's
     // preferred language, default to one that has strings.
     language = 'en';
   }
   const match = useRouteMatch();
   const { hash, search, pathname } = useLocation();
-
+  useEffect(() => {
+    dynamicActivate(language);
+  }, [language]);
+  i18n.activate(language);
```

Comment:
why do you need this line? ^^

Reply:
@semoal I get an error and it doesn't compile without that
`I18nProvider did not render. A call to i18n.activate still needs to happen or forceRenderOnLocaleChange must be set to false.`

Reply:
Because you need to add that to the dynamicActivate function, just behind i18n.load(), extracting this out of the useEffect you're re-running it on each re-render